pkg/service/controller.go

```diff
@@ -52,6 +70,25 @@ func (c *ControllerService) validateCreateVolumeRequest(req *csi.CreateVolumeReq
 		return status.Error(codes.InvalidArgument, "volume capabilities missing in request")
 	}
 
+	csiConfigMap, err := c.infraClientset.CoreV1().ConfigMaps(c.infraClusterNamespace).Get(ctx, configMapName, metav1.GetOptions{})
+	if err != nil {
+		return status.Error(codes.FailedPrecondition, "Couldn't find a configmap for the csi driver")
+	}
+
+	var allowedStorageClassesArr []string
+	allowedStorageClasses, ok := csiConfigMap.Data[allowedInfraStorageclassesKey] //Todo: check if key exists
+	if !ok {
+		allowedStorageClassesArr = defaultAllowedInfraStorageclasses
+	} else {
+		allowedStorageClassesArr = strings.Split(allowedStorageClasses, ",")
+	}
+
+	storageClassName := req.Parameters[InfraStorageClassNameParameter]
+
+	if !contains(allowedStorageClassesArr, storageClassName) {
+		return status.Error(codes.InvalidArgument, "infraStorageclass is not in the allowed list")
+	}
```

Comment:
what should be the behavior if no `allowedStroageclasses` arg exists? I think we should default to allowing all storageclasses unless this argument is set.